scripts/build-wda.js

```diff
@@ -1,27 +1,38 @@
 const {WebDriverAgent} = require('appium-webdriveragent');
 const xcode = require('appium-xcode');
-const B = require('bluebird');
 const {Simctl} = require('node-simctl');
 const {getSimulator} = require('appium-ios-simulator');
 const {logger} = require('appium/support');
 
 const log = logger.getLogger('WDA');
 
-// TODO: allow passing in all the various build params as CLI args
+function parseArgValue(argName) {
+  const argNamePattern = new RegExp(`^--${argName}\\b`);
+  for (let i = 1; i < process.argv.length; ++i) {
+    const arg = process.argv[i];
+    if (argNamePattern.test(arg)) {
+      return arg.includes('=') ? arg.split('=')[1] : process.argv[i + 1];
+    }
+  }
+  return null;
+}
+
 async function build() {
-  const [xcodeVersion, platformVersion] = await B.all([
-    xcode.getVersion(true),
-    xcode.getMaxIOSSDK(),
-  ]);
+  const customDevice = parseArgValue('name');
+  const xcodeVersion = await xcode.getVersion(true);
+  const platformVersion = parseArgValue('sdk') || (await xcode.getMaxIOSSDK());
+
   const verifyDevicePresence = (info) => {
     if (!info) {
-      throw new Error(`Cannot find any available iOS ${platformVersion} Simulator on your system`);
+      throw new Error(
+        `Cannot find any available iOS ${platformVersion} ${customDevice || ''} Simulator on your system`,
```

Comment:
Please tune the formatting so there is no double space if customDevice is empty

Reply:
done